pages/common/aws-dynamodb.md

```diff
@@ -0,0 +1,36 @@
+# aws dynamodb
+
+> CLI for AWS dynamodb.
+> More information: <https://docs.aws.amazon.com/cli/latest/reference/dynamodb/>.
+
+- Create a table:
+
+`aws dynamodb create-table --table-name {{TableName}} --attribute-definitions AttributeName=S,AttributeType=S --key-schema AttributeName=S,KeyType=HASH --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5`
+
+- List all tables in the DynamoDB:
+
+`aws dynamodb list-tables`
+
+- Get details about a specific table:
+
+`aws dynamodb describe-table --table-name {{TableName}}`
+
+- Add an item to a table:
+
+`aws dynamodb put-item --table-name {{TableName}} --item '{"AttributeName": {"S": "value"}}'`
+
+- Retrieve an item from a table:
+
+`aws dynamodb get-item \ --table-name {{TableName}} \ --key '{"ID": {"N": "1"}}'`
+
+- Update an item in the table:
+
+`aws dynamodb update-item --table-name {{TableName}} --key '{"ID": {"N": "1"}}' --update-expression "SET Name = :n" --expression-attribute-values '{":n": {"S": "Jane"}}'`
+
+- Scan items in the table:
+
+`aws dynamodb scan --table-name {{TableName}}`
+
+- Delete an item from the table:
+
+`aws dynamodb delete-item --table-name {{TableName}} --key '{"ID": {"N": "1"}}'`
```

Comment:
Thanks for your contribution, I have some suggestions for this page.

```suggestion
`aws dynamodb create-table --table-name {{table_name}} --attribute-definitions {{AttributeName=S,AttributeType=S}} --key-schema {{AttributeName=S,KeyType=HASH}} --provisioned-throughput {{ReadCapacityUnits=5,WriteCapacityUnits=5}}`

- List all tables in the DynamoDB:

`aws dynamodb list-tables`

- Get details about a specific table:

`aws dynamodb describe-table --table-name {{table_name}}`

- Add an item to a table:

`aws dynamodb put-item --table-name {{table_name}} --item '{{"AttributeName": {"S": "value"}}}'`

- Retrieve an item from a table:

`aws dynamodb get-item --table-name {{table_name}} --key '{{{"ID": {"N": "1"}}}'`

- Update an item in the table:

`aws dynamodb update-item --table-name {{table_name}} --key '{{{"ID": {"N": "1"}}}}' --update-expression "{{SET Name = :n}}" --expression-attribute-values '{{{":n": {"S": "Jane"}}}}'`

- Scan items in the table:

`aws dynamodb scan --table-name {{table_name}}`

- Delete an item from the table:

`aws dynamodb delete-item --table-name {{table_name}} --key '{{{"ID": {"N": "1"}}}}'`
```

We use `{{}}` to highlight placeholders with a different style/color in our clients. In this case, since DynamoDB uses it too, I have manually tested the placeholders locally for highlighting before making this suggestion.

![image](https://github.com/tldr-pages/tldr/assets/26346867/c22188ae-6b6d-4b97-9712-ceb1dcdf46ee)